kernel-spark/src/main/java/io/delta/kernel/spark/catalog/SparkTable.java

```diff
@@ -169,7 +168,7 @@ public ScanBuilder newScanBuilder(CaseInsensitiveStringMap scanOptions) {
     Map<String, String> combined = new HashMap<>(this.options);
     combined.putAll(scanOptions.asCaseSensitiveMap());
     CaseInsensitiveStringMap merged = new CaseInsensitiveStringMap(combined);
-    return new SparkScanBuilder(name(), tablePath, dataSchema, partitionSchema, snapshot, merged);
+    return new SparkScanBuilder(name(), snapshotManager, dataSchema, partitionSchema, merged);
```

Comment:
since the snapshot is already in SparkTable, why passing down the `snapshotManager`?

Reply:
`MicroBatchSteam` will use it for refreshing latest snapshot during execution. `snapshot` will just be the initial snapshot for streaming queries